src/js/utils/string-utils.js

```diff
@@ -16,19 +16,18 @@
          * indicated via curly braces, e.g. 'Hello {name}'. These tokens are
          * replaced by the parameter value of the same name.
          *
-         * Parameter values will be rendered using their toString methods and then
-         * HTML-escaped. The only exception is that instances of the class HTML
-         * are rendered without escaping as their contract declares that they are
-         * already valid HTML.
+         * Parameter values will be rendered using their toString methods.
+         * **NO** HTML escaping or sanitizing of any form is performed.
+         * If HTML escaping is required, use HtmlUtils.interpolateHtml().
          *
          * Example:
          *
          *~~~ javascript
-         * HtmlUtils.interpolate(
+         * StringUtils.interpolate(
          *     'You are enrolling in {spanStart}{courseName}{spanEnd}',
          *     {
          *         courseName: 'Rock & Roll 101',
-         *         spanStart: HtmlUtils.HTML('<span class="course-title">'),
+         *         spanStart: '<span class="course-title">',
          *         spanEnd: HtmlUtils.HTML('</span>')
```

Comment:
I agree that your change here matches the [docs](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/preventing_xss/preventing_xss.html#legacy-javascript-files), but this is a bad example. I guess someone copy-pasted the docs from `HtmlUtils.interpolate` to become the docs to `StringUtils.interpolate` and didn't update it.

If I'm understanding this correctly: In the case of `StringUtils.interpolate`, it wouldn't make any sense to deliberately pass in some `HtmlUtils.HTML` as parameters, because the output of `StringUtils.interpolate` is a plain string, and the `HtmlUtils.HTML` safety of the HTML will get lost, resulting in that HTML being escaped anyways.

The example output below, `'You are enrolling in <span class="course-title">Rock & Roll 101</span>'`, while correct, seems to imply that the HTML is desired and that the whole string would somehow be passed to an end-user as-is, which is a security issue because it contains unescaped user input. Or, if this was properly escaped later ("delaying escaping as late as possible"), the result that users see would be `You are enrolling in &lt;span class="course-title"&gt;Rock &amp; Roll 101&lt;/span&gt;` which seems clearly wrong.

A better example would just be

```
StringUtils.interpolate(
    'You are enrolling in {courseName}',
    {courseName: 'Rock & Roll 101'},
);
```

with the output `"You are enrolling in Rock & Roll 101"; // plain string, unescaped`

Reply:
@bradenmacdonald @robrap yep fair enough, I see what you're saying. My intention for showing those in the examples was to be absolutely clear what does html escaping and what does not. I'll change them though

Reply:
@bradenmacdonald also your example isn't a great real-world example, because no-one should actually do that with escaping `courseName`.

Reply:
@robrap  @bradenmacdonald please see the changes I made - I followed your suggestions, and added more examples for clarity.

Reply:
👍 LGTM.